### ISSUE TITLE
Delete records in partial Voyager Pipeline

### ISSUE THREAD

**Issue by patrickzurek** [eXtensibleCatalog/Drupal-Toolkit#116]
JIRA issue created by: rcook
Originally opened: 2012-07-17 11:02 AM

Issue body:

I have been getting frustrated that for several weeks we failed the validation tests when harvesting a delete from the partial Voyager pipeline.

Now MT thinks he is ready to test this but there seem to be no deletes.  Is that true or has a step in the process failed?  Ralph, can you check?

The most critical thing for Drupal right now is to be able to run it in production daily update mode.  How can we move forward with this?

[~sranganathan] [~mwesley]

**Comment by patrickzurek** [comment]
JIRA Comment by user: rcook
JIRA Timestamp: 2012-07-17 11:06 AM

Comment body:

Note that this server has been up since May 18 and has only processed a handful of deletes, and nothing after about June 14th.

**Comment by patrickzurek** [comment]
JIRA Comment by user: rcook
JIRA Timestamp: 2012-07-17 11:15 AM

Comment body:

If I am correct and we have not processed some # of weeks of deletes, then I would like to schedule (between Ralph and MT) a special processing of the deletes, so we can walk through the validation of those.  If there are several weeks of files behind, then I would like us to hand process each of those, one at a time, with a validation step in between.

**Comment by patrickzurek** [comment]
JIRA Comment by user: rcook
JIRA Timestamp: 2012-07-17 11:25 AM

Comment body:

[~mwesley] Ralph has many weeks of deletes that have not been processed.  If you are sure you match, then he will process a file today.

**Comment by patrickzurek** [comment]
JIRA Comment by user: mwesley
JIRA Timestamp: 2012-07-17 01:57 PM

Comment body:

Yes, go ahead and process a file for today.

**Comment by patrickzurek** [comment]
JIRA Comment by user: admin
JIRA Timestamp: 2012-07-17 02:33 PM

Comment body:

I usually process two files at a time - one for Bibs and one for MFHDs. Should I process both of them, or just pick one?

**Comment by patrickzurek** [comment]
JIRA Comment by user: rcook
JIRA Timestamp: 2012-07-17 02:46 PM

Comment body:

Let's do the "real" scenario and do both, unless MT objects.

**Comment by patrickzurek** [comment]
JIRA Comment by user: admin
JIRA Timestamp: 2012-07-17 03:29 PM

Comment body:

Ok, I processed one bib and one mfhd:
[LIB] Import statistics summary: created 0, updated: 0, skipped: 11261, invalid: 0, deleted: 49, bib: 0, auth: 0, holdings: 0 records. Invalid files: 0. It took 00:00:15.471. checkTime: 00:00:00.204. insertTime: 00:00:00.046. others: 00:00:15.221

**Comment by patrickzurek** [comment]
JIRA Comment by user: mwesley
JIRA Timestamp: 2012-07-17 09:01 PM

Comment body:

Harvest completed a few minutes ago and deletes went through well. Drupal statistics still match MST logs. 108 deleted XC records were harvested, although none of them were active. They all were "upd_del_prev_del_cnt". We can do the next set of deletes on Wednesday.

**Comment by patrickzurek** [comment]
JIRA Comment by user: admin
JIRA Timestamp: 2012-07-18 09:13 AM

Comment body:

Processed another set of deletes this morning:
2012-07-18 08:47:16,439 [main](Importer.java:808) INFO  - [LIB] Import statistics summary: created 0, updated: 0, skipped: 669, invalid: 0, deleted: 41, bib: 0, auth: 0, holdings: 0 records. Invalid files: 0. It took 00:00:01.654. checkTime: 00:00:00.077. insertTime: 00:00:00.761. others: 00:00:00.816

**Comment by patrickzurek** [comment]
JIRA Comment by user: rcook
JIRA Timestamp: 2012-07-18 11:42 AM

Comment body:

This not a good approach.   We need to test deletes in a real way.  We have too small a cross section of Voyager records loaded in the partial pipeline that it could be weeks before we "touch" one of our loaded records.   I think we have to move to testing this with a full load of data records.  [~mwesley] do you have a place to proceed with this test?  do you need anything from [~bgant] or [~admin]?

Thoughts?

**Comment by patrickzurek** [comment]
JIRA Comment by user: mwesley
JIRA Timestamp: 2012-07-19 05:55 AM

Comment body:

I can use the same 226 server. So I will set up another Drupal Toolkit instance. However, can we continue this method for the rest of the week. Today, the last harvest brought in deletes for existing metadata in the DT. The count is off by 24 Manifestations (DT count is too high). Likewise, nodes are off by 24. Seems to me as if 24 Manifestations did not get deleted in the DT. There was also a PHP fatal error that shut down cron due to a rouge debug function. I got rid of the function, but for the records not being deleted, I'm taking a look at why this happened and will put together a fix.

**Comment by patrickzurek** [comment]
JIRA Comment by user: rcook
JIRA Timestamp: 2012-07-19 10:03 AM

Comment body:

Hooray!  I have known for many months that the deletes are not handled correctly.  I hope now that we have a targeted batch to dig into you can figure this out.    It is THE single biggest issue holding us back (in my opinion).

**Comment by patrickzurek** [comment]
JIRA Comment by user: admin
JIRA Timestamp: 2012-07-19 10:20 AM

Comment body:

Today's deletes:
2012-07-19 10:01:32,971 [main](Importer.java:808) INFO  - [LIB] Import statistics summary: created 0, updated: 0, skipped: 3378, invalid: 0, deleted: 25, bib: 0, auth: 0, holdings: 0 records. Invalid files: 0. It took 00:00:05.987. checkTime: 00:00:00.125. insertTime: 00:00:03.323. others: 00:00:02.539

**Comment by patrickzurek** [comment]
JIRA Comment by user: mwesley
JIRA Timestamp: 2012-07-30 04:57 AM

Comment body:

I've tested the partial voyager set enough. I have a good idea of what's going wrong and have made a few fixes. Will want to test on the full Voyager set.

I'm moving everything to a single issue: Harvested records match MST at #DRUPAL-98 – from months of testing, I've concluded that the problem is with the harvested deletes of manifestation records not being deleted in Drupal after the harvest – although they are clearly marked for deletion. A fix is in the works.
